CONTRIBUTING.md

```diff
@@ -0,0 +1,131 @@
+
+There's a module called `util/preconditions`, loosely based on `preconditions.js`, based on `guava`, that we use for state and argument checking. It should be trivial to use. We recommend using it on all methods, in order to improve robustness and consistency.
+
+```javascript
+$.checkState(something === anotherthing, 'Expected something to be anotherthing');
+$.checkArgument(something < 100, 'something', 'must be less than 100');
+$.checkArgumentType(something, PrivateKey, 'something'); // The third argument is a helper to mention the name of the argument
+$.checkArgumentType(something, PrivateKey); // but it's optional (will show up as "(unknown argument)")
+```
+
+### Testing
+
+#### T1 - Tests Must be Written Elegantly
+
+Style guidelines are not relaxed for tests. Tests are a good way to show how to use the library, and maintaining them is extremely necessary.
+
+Don't write long tests, write helper functions to make them be as short and concise as possible (they should take just a few lines each), and use good variable names.
+
+#### T2 - Tests Must not be Random
+
+Inputs for tests should not be generated randomly. Also, the type and structure of outputs should be checked.
+
+#### T3 - Data for Tests Included in a JSON File
+
+If possible, data for tests should be included in a JSON file in the `test/data` directory. This improves interoperability with other libraries and keeps tests cleaner.
+
+## Pull Request Workflow
+
+Our workflow is based on GitHub's pull requests. We use feature branches, prepended with: `test`, `feature`, `fix`, `refactor`, or `remove` according to the change the branch introduces. Some examples for such branches are:
+```sh
+git checkout -b test/some-module
+git checkout -b feature/some-new-stuff
+git checkout -b fix/some-bug
+git checkout -b remove/some-file
+```
+
+We expect pull requests to be rebased to the master branch before merging:
+```sh
+git remote add bitpay git@github.com:bitpay/bitcore-node.git
+git pull --rebase bitpay master
+```
+
+Note that we require rebasing your branch instead of merging it, for commit readability reasons.
+
+After that, you can push the changes to your fork, by doing:
+```sh
+git push origin your_branch_name
+git push origin feature/some-new-stuff
+git push origin fix/some-bug
+```
+Finally go to [github.com/bitpay/bitcore-node](https://github.com/bitpay/bitcore-node) in your web browser and issue a new pull request.
+
+Main contributors will review your code and possibly ask for changes before your code is pulled in to the main repository.  We'll check that all tests pass, review the coding style, and check for general code correctness. If everything is OK, we'll merge your pull request and your code will be part of bitcore.
+
+If you have any questions feel free to post them to
+[github.com/bitpay/bitcore-node/issues](https://github.com/bitpay/bitcore-node/issues).
+
+Thanks for your time and code!
```

Comment:
I removed several rules from the original file, mostly those focused on *developing a library*.  I believe we should write a different guideline focusing on how to *build solid infrastructure*.

This two files may be very similar at the moment, but I expect this to change soon.